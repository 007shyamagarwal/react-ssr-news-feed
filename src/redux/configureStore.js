import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';
import { composeWithDevTools } from 'redux-devtools-extension';

const localStorageMiddleware = ({ getState }) => {
  return (next) => (action) => {
    const result = next(action);
    localStorage.setItem(
      `applicationState/${getState().pageId}`,
      JSON.stringify(getState())
    );
    return result;
  };
};

const reHydrateStore = (isServer, pageId) => {
  if (isServer) return;
  if (localStorage.getItem(`applicationState/${pageId}`) !== null) {
    return JSON.parse(localStorage.getItem(`applicationState/${pageId}`)); // re-hydrate the store
  }
};

export default function configureStore(preloadedState, isServer) {
  const { pageId } = preloadedState;
  const middlewares = isServer
    ? [thunkMiddleware]
    : [thunkMiddleware, localStorageMiddleware];
  return createStore(
    rootReducer,
    reHydrateStore(isServer, pageId) || preloadedState,
    composeWithDevTools(applyMiddleware(...middlewares))
  );
}
