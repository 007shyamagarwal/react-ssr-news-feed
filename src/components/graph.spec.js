import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import MyChart from './graph';

describe('MyChart', () => {
  const initialState = {
    isFetching: false,
    posts: { hits: [] },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('MyChart page with empty content', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyChart />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
