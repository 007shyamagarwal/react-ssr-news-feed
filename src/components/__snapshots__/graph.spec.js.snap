// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyChart MyChart page with content 1`] = `
<div
  style={
    Object {
      "height": "300px",
      "marginTop": "30px",
      "width": "100%",
    }
  }
>
  <div>
    Loading Chart
  </div>
</div>
`;

exports[`MyChart MyChart page with empty content 1`] = `
<div
  style={
    Object {
      "height": "300px",
      "marginTop": "30px",
      "width": "100%",
    }
  }
>
  <div>
    Loading Chart
  </div>
</div>
`;
